src/main/java/co/rsk/federate/signing/hsm/message/AdvanceBlockchainMessage.java

```diff
@@ -23,4 +27,22 @@ private String parseBlockHeader(BlockHeader blockHeader) {
     public List<String> getBlockHeaders() {
         return blockHeaders;
     }
+
+    public List<String[]> getBrothersList() {
+        return brothersList;
+    }
+
+    private List<String[]> parseBrothersBlockHeaders(List<List<BlockHeader>> brothersList) {
+        List<String[]> result = new ArrayList<>();
+        for (List<BlockHeader> headers : brothersList) {
+            String[] blockHeadersArray = new String[headers.size()];
+            int headersIndex = 0;
+            // Invert order
+            for (int i = headers.size() - 1; i >= 0; i--) {
+                blockHeadersArray[headersIndex++] = parseBlockHeader(headers.get(i));
+            }
+            result.add(blockHeadersArray);
+        }
+        return result;
```

Comment:
Should the list of brothers also be reversed so that brothersList.get(0) has the list of uncles that correspond to blockHeaders[0]?

As far as I see, this method is only reversing the list of brothers inside each element of brothersList. Why is that by the way?

Reply:
No , The List Of Brothers Need Not Be Reversed, Only Its Block Headers.